### PR TITLE
ROCm: Fix linking of custom ops in load_inline

### PR DESCRIPTION
**Pull request by t-vi**
Previously we did not link against amdhip64 (roughly equivalent to cudart). Apparently, the recent RTDL_GLOBAL fixes prevent the extensions from finding the symbols needed for launching kernels.

### CODE REVIEW

**Comment by t-vi** [comment]
@jeffdaily

**Comment by jeffdaily** [comment]
This library is new as of ROCm 3.5.  Can we version guard these changes to preserve some backward compatibility?

**Comment by t-vi** [comment]
@jeffdaily Sure, what do we need to link for older ROCm to be able to launch kernels?

File: test/test_cpp_extensions_jit.py [hunk 1]
```diff
@@ -327,6 +327,53 @@ def test_inline_jit_compile_extension_cuda(self):
         z = module.cos_add(x, y)
         self.assertEqual(z, x.cos() + y.cos())
 
+    @unittest.skipIf(not TEST_CUDA, "CUDA not found")
```
Comment (jeffdaily) [inline test/test_cpp_extensions_jit.py:330]: Should we add the `not (TEST_CUDA or TEST_ROCM)` in this PR, or do you want to create a follow-up PR to enable many of these `TEST_CUDA`-only tests?

Comment (jeffdaily) [reply test/test_cpp_extensions_jit.py:330]: Also I should note that the test script where you've added this test is one of the "slow" ones that does not run automatically in CI.

**Comment by t-vi** [comment]
Let's enable them all and see how it goes.

File: test/test_cpp_extensions_jit.py (hunk 1 above)
Comment (t-vi) [reply test/test_cpp_extensions_jit.py]: Let's enable them all. Not at all or not run on all instances?

Comment (t-vi) [reply test/test_cpp_extensions_jit.py]: Next, I'll try a non-stupid variant...

**Comment by ezyang** [comment]
doesn't seem to be enough, rocm test still failing

**Comment by t-vi** [comment]
Yes, I'm at a loss why it happens, it does not on my ROCm box.

**Comment by jeffdaily** [comment]
It doesn't happen on my local box, either.  However, the missing symbol that CI is complaining about is in libhip_hcc.so.  Should you try adding that for ROCm <= 3.3?

**Comment by t-vi** [comment]
Ha, I'll try that. Thank you!

**Comment by t-vi** [comment]
Now there is a bad interaction with a new patch, as we hit this line:

https://github.com/pytorch/pytorch/blob/288ece89e16fe915ae0cfb2d948447e90ab03b1a/torch/testing/_internal/common_cuda.py#L47

I'll send a fix.

**Comment by t-vi** [comment]
@ezyang @jeffdaily I'd claim that it's working now and the windows test failure is from master.

**Review by jeffdaily — APPROVED** [review]
LGTM.  Does test_cpp_extensions_jit.py run during CI?

**Comment by t-vi** [comment]
@jeffdaily 
> Does test_cpp_extensions_jit.py run during CI?

Yes, in the test2 on pytorch-linux-xenial-rocm3.5.1-py3.6 it is around (in plain text time or at least system time), there is:
12:40:11  test_inline_jit_compile_custom_op_cuda (__main__.TestCppExtensionJIT)
in this file:
https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-linux-xenial-rocm3.5.1-py3.6-test2/201/timestamps/?time=HH:mm:ss&appendLog&locale=en_US

**Comment by jeffdaily** [comment]
This PR #40800 added the `torch.cuda.version()` that you had to fix when you rebased.  It got reverted since that change broke ROCm.  Your fix was probably sufficient, but the entire PR got reverted, so now you need to rebase this PR again.

**Comment by t-vi** [comment]
No, I'll stop here, really.

**Comment by t-vi** [comment]
I'd be keen to get this in before it breaks again... :wink:

**Comment by t-vi** [comment]
@ezyang If you have a hint what I need to check from the failed FB internal test...?

**Pull request closed without merging**